The given compilation instructions for numba-dpex doesn't work

If I follow the `numba-dpex` compilation instructions according to `dpbench`'s [readme](https://github.com/IntelPython/dpbench/blob/main/README.md), it doesn't work. I get this error:

```bash
python setup.py develop
Compiling numba_dpex/dpnp_iface/dpnp_fptr_interface.pyx because it changed.
[1/1] Cythonizing numba_dpex/dpnp_iface/dpnp_fptr_interface.pyx
running develop
/localdisk/work/$USER/.dpbench/lib/python3.9/site-packages/setuptools/command/easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
/localdisk/work/$USER/.dpbench/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
running egg_info
creating numba_dpex.egg-info
writing numba_dpex.egg-info/PKG-INFO
writing dependency_links to numba_dpex.egg-info/dependency_links.txt
writing entry points to numba_dpex.egg-info/entry_points.txt
writing requirements to numba_dpex.egg-info/requires.txt
writing top-level names to numba_dpex.egg-info/top_level.txt
writing manifest file 'numba_dpex.egg-info/SOURCES.txt'
reading manifest file 'numba_dpex.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
adding license file 'LICENSE'
writing manifest file 'numba_dpex.egg-info/SOURCES.txt'
running build_ext
building 'numba_dpex._usm_allocators_ext' extension
creating build
creating build/temp.linux-x86_64-3.9
creating build/temp.linux-x86_64-3.9/numba_dpex
creating build/temp.linux-x86_64-3.9/numba_dpex/dpctl_iface
gcc -pthread -B /localdisk/work/$USER/.dpbench/compiler_compat -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /localdisk/work/$USER/.dpbench/include -I/localdisk/work/$USER/.dpbench/include -fPIC -O2 -isystem /localdisk/work/$USER/.dpbench/include -fPIC -I/localdisk/work/$USER/.dpbench/lib/python3.9/site-packages -I/localdisk/work/$USER/.dpbench/lib/python3.9/site-packages/dpctl/include -I/localdisk/work/$USER/.dpbench/include/python3.9 -c numba_dpex/dpctl_iface/usm_allocators_ext.c -o build/temp.linux-x86_64-3.9/numba_dpex/dpctl_iface/usm_allocators_ext.o
In file included from /localdisk/work/$USER/.dpbench/lib/python3.9/site-packages/dpctl/include/dpctl_sycl_interface.h:35,
                 from numba_dpex/dpctl_iface/usm_allocators_ext.c:35:
/localdisk/work/$USER/.dpbench/lib/python3.9/site-packages/dpctl/include/syclinterface/dpctl_sycl_device_interface.h:253:1: error: expected identifier or ‘(’ before ‘[’ token
  253 | [[deprecated("Use DPCTLDevice_WorkItemSizes3d instead")]] DPCTL_API
      | ^
error: command '/usr/bin/gcc' failed with exit code 1
```

However, if I follow the instructions from the `numba-dpex` page, like this:

```bash
conda create -n dpbench-dev -c /opt/intel/oneapi/conda_channel python=3.9 dpctl dpnp numba spirv-tools llvm-spirv llvmdev cython pytest
conda activate dpbench-dev
python setup.py develop
```

It works. 

I think the compilation instructions on `dpbench` readme needs to be changed.

--- a/.github/workflows/build_and_run.yml
+++ b/.github/workflows/build_and_run.yml
@@ -30,7 +30,7 @@ jobs:
         with:
           auto-update-conda: true
           python-version: ${{ matrix.python }}
-          activate-environment: dpbench-devel
+          activate-environment: dpbench-dev
 
       - name: Conda info
         shell: bash -l {0}
@@ -39,10 +39,10 @@ jobs:
       - name: Install dpbench dependencies
         shell: bash -l {0}
         run: |
-          conda install tbb=2021.6.0 dpcpp_linux-64 -c intel
+          conda install -c intel tbb=2021.6.0 dpcpp_linux-64
           conda install numpy numba cython cmake ninja scikit-build pandas
           conda install scipy spirv-tools scikit-learn pybind11
-          conda install libgcc-ng>=11.2.0 libstdcxx-ng>=11.2.0 libgomp>=11.2.0 -c pkgs/main
+          conda install -c pkgs/main libgcc-ng">=11.2.0" libstdcxx-ng">=11.2.0" libgomp">=11.2.0"
           conda install -c dppy/label/dev -c intel dpctl=0.13.0 numba-dpex=0.18.1 dpnp=0.10.1
           conda list
 

--- a/README.md
+++ b/README.md
@@ -10,15 +10,17 @@
 ## Examples of setting up and running the benchmarks
 1. Setting up conda environment and installing dependencies:
 
-        $ conda create -n dpbench
-        $ conda activate dpbench
-        $ conda install python=3.9 numpy cython cmake ninja scikit-build conda-forge::gtest conda-forge::gmock pytest
-        $ conda install dpnp -c dppy/label/dev -c /opt/intel/oneapi/conda_channel --override-channels
-        $ conda install numba scipy scikit-learn spirv-tools
-        $ conda install pybind11
-        $ conda install dpcpp_linux-64 -c intel
+        $ conda create -n dpbench-dev
+        $ conda activate dpbench-dev
+        $ conda install python=3.9
+        $ conda install -c intel tbb=2021.6.0 dpcpp_linux-64
+        $ conda install numpy numba cython cmake ninja scikit-build pandas
+        $ conda install scipy spirv-tools scikit-learn pybind11
+        # do not miss the quotes!
+        $ conda install -c pkgs/main libgcc-ng">=11.2.0" libstdcxx-ng">=11.2.0" libgomp">=11.2.0"
+        $ conda install -c dppy/label/dev -c intel dpctl=0.13.0 numba-dpex=0.18.1 dpnp=0.10.1
 
-2. Build Numba-Dpex
+2. Build Numba-Dpex (If you want to use your own numba-dpex)
 
         $ git clone https://github.com/IntelPython/numba-dpex.git
         $ cd numba-dpex/
@@ -26,17 +28,26 @@
         $ cd ..
 
 3. Build and run DPBench
-
-        #To build:
+    - To build:
+        ```bash
         $  CC=icx CXX=icpx python setup.py develop -- -Dpybind11_DIR=$(python -m pybind11 --cmakedir) -DDPCTL_MODULE_PATH=$(python -m dpctl --cmakedir)
-        #To run, taking black_scholes for example:
+        ```
+    - To run, taking black_scholes for example:
+        ```bash
         $  python -c "import dpbench; dpbench.run_benchmark(\"black_scholes\")"
+        ```
 
 4. Device Customization
 
-        #Device can be selected via providing customized arch in framework Json file. Here is an example:
+   If a framework is SYCL based, an extra configuration option `sycl_device` may be set in the
+   framework JSON file to control what device the framework uses for execution. The `sycl_device`
+   value should be a legal
+   [SYCL device filter ](https://intel.github.io/llvm-docs/EnvironmentVariables.html#sycl_device_filter)
+   string. The dpcpp, dpnp, numba_dpex_kernel, and numba_dpex frameworks support the sycl_device option.
 
-```json
+   Here is an example:
+
+    ```json
         {
             "framework": {
                 "simple_name": "dpcpp",
@@ -44,11 +55,17 @@
                 "prefix": "dp",
                 "postfix": "dpcpp",
                 "class": "DpcppFramework",
-                "arch": "gpu"
+                "arch": "gpu",
+                "sycl_device": "level_zero:gpu:0"
             }
         }
-```
+    ```
 
-        #To run with custimized Json file:
+    > **_NOTE:_**  The `arch` option is deprecated and not used by dpbench.
 
-        $ python -c "import dpbench; dpbench.run_benchmark(\"black_scholes\", "<absolute path to json file>")"
+   To run with custimized framework JSON file, pass it as an argument to the `run_benchmark` or
+   `run_benchmarks` functions.
+
+    ```bash
+    $ python -c "import dpbench; dpbench.run_benchmark(\"black_scholes\", "<absolute path to json file>")"
+    ```

